dockerGrader/testServer.py

```diff
@@ -0,0 +1,121 @@
+import base64, boto3, botocore, logging, os, sys, json, subprocess, shutil
+from flask import Flask
+
+ACCESS_PATH = "projects/{project}/users/{googleId}/curr.json"
+CODE_DIR = "./submission"
+TEST_DIR = "/tmp/test"
+
+SUBMISSIONS = 'submissions'
+BUCKET = 'caraza-harter-cs301'
+session = boto3.Session(profile_name='cs301ta')
+s3 = session.client('s3')
+
+app = Flask(__name__)
+
+logging.basicConfig(
+    handlers=[
+        logging.FileHandler("testServer.log"),
+        logging.StreamHandler()
+    ],
+    level=logging.WARNING)
+
+def downloadSubmission(projectPath):
+    # a project path will look something like this:
+    # projects/p0/users/115799594197844895033/curr.json
+
+    userId = projectPath.split('/')[-2]
+
+    # create user dir for download
+    logging.info('download to {}'.format(CODE_DIR))
+    if os.path.exists(CODE_DIR):
+        shutil.rmtree(CODE_DIR)
+    os.mkdir(CODE_DIR)
+
+    # download
+    response = s3.get_object(Bucket=BUCKET, Key=projectPath)
+    submission = json.loads(response['Body'].read().decode('utf-8'))
+    fileContents = base64.b64decode(submission.pop('payload'))
+    with open(os.path.join(CODE_DIR, submission['filename']), 'wb') as f:
+        f.write(fileContents)
+    with open(os.path.join(CODE_DIR, 'meta.json'), 'w') as f:
+        f.write(json.dumps(submission, indent=2))
+
+def uploadResult(project, studentID):
```

Comment:
don't you need to call this after you run the tests?

Reply:
Oh... I uploaded a wrong version. (JSON processing function is not as same as my latest version) I will update in the next commit.